ONSdigital/eq-author-app
# Add refetch and correct error conditions

> BEFORE MAKING YOUR PR... There must be no linting errors and all tests must pass

### What is the context of this PR?

Make the decimal error message appear straight away instead of waiting for a refresh.

This was done by adding a graphql query refetch on answer update and correcting the conditions for adding the decimal error message to the error validation array.

It was just printing if it wasn't `null` but wasn't checking for `undefined`. 
It also was firing if the values were just not equal

This works but I'm wary of adding the refetch in case it bogs Author down with network requests.

### How to review 
- Create a questionnaire
- Add two question pages
- First page add a **Number/Currency/Unit/Percentage** answer and set the decimal value to **2** 
- Second page add a **Number/Currency/Unit/Percentage** answer and set the **minimum** value to the Previous answer
    - **Answer type must match**
- ~~Validation error under the decimal box reads - "Enter a decimal that is the same as the associated question page."~~
- Validation error should appear before pressing done on the validation modal

### What to do after everything is green

1. * [ ] Bring this branch up-to-date with Origin/Master
2. * [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. * [ ] Click the **Merge** button on this pull request
4. * [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. * [ ] Move the Trello card for this task into the next stage of the process
